### PR TITLE
master/worker/task add strictly config verification

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Make sure all items in configuration file has been parsed and user always knows the exactly configuration about master worker and task.

### What is changed and how it works?
-  For master worker(.toml) config, check whether undecoded item exists, and if exists, refuse to start server
-  For task(.yaml) config, use `unmarshalStrict` to decode and return error to caller when not all config item decoded and duplicate item existed
-  Add test 
-  Remove useless config item in unit tests and integration tests

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: dm/config/task.go [hunk 1]
```diff
@@ -294,7 +294,7 @@ func (c *TaskConfig) DecodeFile(fpath string) error {
 		return errors.Annotatef(err, "read config file %v", fpath)
 	}
 
-	err = yaml.Unmarshal(bs, c)
+	err = yaml.UnmarshalStrict(bs, c)
```
[the remark this answers is not in the document]
Comment (3pointer) [reply dm/config/task.go]: Done

**Comment by 3pointer** [comment]
> Besides after this pr, old task config or DM instance config in user environment may not work because of historical deprecated configuration, we'd better update docs to record this after this pr is merged. https://github.com/pingcap/docs-cn/blob/master/dev/reference/tools/data-migration/dm-upgrade.md

OK, And maybe ansible-playbook config templates also need to update? but I didn't find dm-ansible git repo.

**Comment by amyangfei** [comment]
> OK, And maybe ansible-playbook config templates also need to update? but I didn't find dm-ansible git repo.

it is in DM repo, `dm/dm-ansible` directory

**Comment by 3pointer** [comment]
> > OK, And maybe ansible-playbook config templates also need to update? but I didn't find dm-ansible git repo.
> 
> it is in DM repo, `dm/dm-ansible` directory

I have removed `meta-file` in dm-worker template in latest commit, other useless config seem to be removed already, please review it

**Pull request merged**